src/azure-cli/azure/cli/command_modules/appconfig/_validators.py

```diff
@@ -159,3 +162,32 @@ def validate_secret_identifier(namespace):
         KeyVaultIdentifier(uri=identifier)
     except Exception as e:
         raise CLIError("Received an exception while validating the format of secret identifier.\n{0}".format(str(e)))
+
+
+def validate_key(namespace):
+    if namespace.key:
+        input_key = str(namespace.key)
+        if input_key == '.' or input_key == '..' or '%' in input_key:
+            raise CLIError("Key is invalid. Key cannot be a '.' or '..', or contain the '%' character.")
+        if input_key.startswith(FEATURE_FLAG_PREFIX):
+            raise CLIError("Key is invalid. Key cannot start with the reserved prefix for feature flags.")
+    else:
+        raise CLIError("Key cannot be empty.")
+
+
+def validate_content_type(namespace):
+    if namespace.content_type is not None:
+        if namespace.content_type == FEATURE_FLAG_CONTENT_TYPE:
+            raise CLIError("Content type is invalid. It's a reserved content type for feature flags.")
```

Comment:
Are we case sensitive here? Or when we retrieve feature flags from config store, does the content has to be identical(case sensitive) with FEATURE_FLAG_CONTENT_TYPE?

Reply:
You're right. I'll make it case insensitive to align with the portal.

---
In reply to: [362634100](https://github.com/Azure/azure-cli/pull/11753#discussion_r362634100) [](ancestors = 362634100)